Assignment 2.Rmd

```diff
@@ -81,20 +101,137 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
+std_actions <- count(EDGE,to)
+tch_actions <- count(EDGE,from)
+
+#make sure variables have proper (mathing) names 
+names(std_actions) <- c("id","n_actions")
+names(tch_actions) <- c("id","n_actions")
 
+#add new variable to student and teacher vectors
+V.STD <- merge(V.STD,std_actions, by = "id")
+V.TCH <- merge(V.TCH,tch_actions, by = "id")
+
+#update VERTEX data with our new variable "n_actions"
+VERTEX <- bind_rows(V.TCH, V.STD)
+
+#update graph data with our new variable "n_actions"
+g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+
+```
+
+
+Now, let's plot!
+```{r}
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count, vertex.size=3*VERTEX$n_actions+10)
+
+```
 
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
 Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
 
+Let's clean up the workspace, get the new data and do some cleaning...
+```{r}
+#clean up and get data
+rm(list = ls())
+D1 <- read.csv("~/R/HUDK/assignment2/hudk4050-classes.csv")
+
+#rename rows with student names and drop student names
+rownames(D1) <- paste(D1$First.Name,D1$Last.Name)
+D1 <- D1[,3:8] 
+
+#standardize space/no space in how subjects are refered to and make it a string
+D1[,1] <- gsub(" ", "", as.character(D1[,1]), fixed = TRUE)
+D1[,2] <- gsub(" ", "", as.character(D1[,2]), fixed = TRUE)
+D1[,3] <- gsub(" ", "", as.character(D1[,3]), fixed = TRUE)
+D1[,4] <- gsub(" ", "", as.character(D1[,4]), fixed = TRUE)
+D1[,5] <- gsub(" ", "", as.character(D1[,5]), fixed = TRUE)
+D1[,6] <- gsub(" ", "", as.character(D1[,6]), fixed = TRUE)
+
+#correct the data of one of our friends(there were only numbers, we need to input letters) 
+D1["ZIMO CHEN",]
+D1["ZIMO CHEN",1] <- gsub(" ", "", paste("HUDK",D1["ZIMO CHEN",1]), fixed = TRUE)
+D1["ZIMO CHEN",2] <- gsub(" ", "", paste("HUDM",D1["ZIMO CHEN",2]), fixed = TRUE)
+D1["ZIMO CHEN",3] <- gsub(" ", "", paste("HUDM",D1["ZIMO CHEN",3]), fixed = TRUE)
+D1["ZIMO CHEN",4] <- gsub(" ", "", paste("HUDM",D1["ZIMO CHEN",4]), fixed = TRUE)
+D1["ZIMO CHEN",]
+
+#also, lets remove HUDK4050 - we already know we are taking that class together (and it would mess up our graphs and our centrality measures!)
+D1[,1][D1[,1] == "HUDK4050"] <- ""
+D1[,2][D1[,2] == "HUDK4050"] <- ""
+D1[,3][D1[,3] == "HUDK4050"] <- ""
+D1[,4][D1[,4] == "HUDK4050"] <- ""
+D1[,5][D1[,5] == "HUDK4050"] <- ""
+D1[,6][D1[,6] == "HUDK4050"] <- ""
```

Comment:
Maybe you can first gather the data. And use simpler code to complete the same task.
eg: try "subset"